src/query/range_query/range_query_u64_fastfield.rs

```diff
@@ -202,12 +217,52 @@ pub mod tests {
     use rand::seq::SliceRandom;
     use rand::SeedableRng;
 
-    use crate::collector::Count;
+    use crate::collector::{Count, TopDocs};
     use crate::query::range_query::range_query_u64_fastfield::FastFieldRangeWeight;
     use crate::query::{QueryParser, Weight};
-    use crate::schema::{NumericOptions, Schema, SchemaBuilder, FAST, INDEXED, STORED, STRING};
+    use crate::schema::{
+        NumericOptions, Schema, SchemaBuilder, FAST, INDEXED, STORED, STRING, TEXT,
+    };
     use crate::{Index, IndexWriter, Term, TERMINATED};
 
+    #[test]
+    fn test_text_field_ff_range_query() -> crate::Result<()> {
+        let mut schema_builder = Schema::builder();
+        schema_builder.add_text_field("title", TEXT | FAST);
+        let schema = schema_builder.build();
+        let index = Index::create_in_ram(schema.clone());
+        let mut index_writer = index.writer_for_tests()?;
+        let title = schema.get_field("title").unwrap();
+        index_writer.add_document(doc!(
+          title => "bbb"
+        ))?;
+        index_writer.add_document(doc!(
+          title => "ddd"
+        ))?;
+        index_writer.commit()?;
+        let reader = index.reader()?;
+        let searcher = reader.searcher();
+        let query_parser = QueryParser::for_index(&index, vec![title]);
+
+        let test_query = |query, num_hits| {
+            let query = query_parser.parse_query(query).unwrap();
+            let top_docs = searcher.search(&query, &TopDocs::with_limit(10)).unwrap();
+            assert_eq!(top_docs.len(), num_hits);
+        };
+
+        test_query("title:[aaa TO ccc]", 1);
+        test_query("title:[aaa TO bbb]", 1);
+        test_query("title:[bbb TO bbb]", 1);
+        test_query("title:[bbb TO ddd]", 2);
+        test_query("title:[bbb TO eee]", 2);
+        test_query("title:[bb TO eee]", 2);
+        test_query("title:[ccc TO ccc]", 0);
+        test_query("title:[ccc TO ddd]", 1);
+        test_query("title:[ccc TO eee]", 1);
```

Comment:
this could use tests for exclusive bounds, and half-unbounded ranges

Reply:
I extended the test for half-unbounded and exclusive ranges